server/src/main/java/org/elasticsearch/index/mapper/DataTierFieldMapper.java

```diff
@@ -1,11 +1,13 @@
 /*
  * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-package org.elasticsearch.xpack.cluster.routing.allocation.mapper;
+package org.elasticsearch.index.mapper;
```

Comment:
I wonder if moving the tier field to core is something that should be done as a prerequisite follow-up. I had almost missed it among other changes here. Was it a requirement for your change?

Reply:
This was needed due to the change in `CoordinatorRewriteContext#getFiledType` https://github.com/elastic/elasticsearch/pull/114990/files#diff-45c9ed5af87237215a5a11f3d2549aa678272c2cbc898b866b43516ec992fab5R86

Reply:
makes sense, we could have probably hacked as well by doing string comparison and creating a custom constant  field type instance? Playing devil's advocate a bit here. Or perhaps do the move as a pre-requisite in a separate PR?